frontend/src/components/elements/ArrowVegaLiteChart/ArrowVegaLiteChart.tsx

```diff
@@ -440,19 +435,47 @@ export function getDataArray(
     const row: { [field: string]: any } = {}
 
     if (hasSupportedIndex) {
-      // eslint-disable-next-line prefer-destructuring
-      row[MagicFields.DATAFRAME_INDEX] = dataProto.index[rowIndex][0]
+      let indexValue = dataProto.getIndexValue(rowIndex, 0)
+      const indexType = dataProto.types.index[0].pandas_type
+      if (typeof indexValue === "bigint") {
+        if (indexType === IndexTypeName.UInt64Index) {
+          indexValue = util.BN.new(bigIntToUint32Array(indexValue))
+        } else if (indexType === IndexTypeName.Int64Index) {
+          indexValue = util.BN.new(bigIntToInt32Array(indexValue))
+        }
+      }
+      row[MagicFields.DATAFRAME_INDEX] = indexValue
     }
 
     for (let colIndex = 0; colIndex < cols; colIndex++) {
-      row[dataProto.columns[0][colIndex]] = dataProto.data[rowIndex][colIndex]
+      let dataValue = dataProto.getDataValue(rowIndex, colIndex)
+      const dataType = dataProto.types.data[colIndex].pandas_type
+      if (typeof dataValue === "bigint") {
+        if (dataType === "int64") {
+          dataValue = util.BN.new(bigIntToInt32Array(dataValue))
+        }
+      }
+      row[dataProto.columns[0][colIndex]] = dataValue
     }
     dataArr.push(row)
   }
 
   return dataArr
 }
 
+function bigIntToInt32Array(num: bigint): Int32Array {
+  const maxInt = BigInt(Number.MAX_SAFE_INTEGER)
+  const excess = num % maxInt
+  const bigPart = num / maxInt
+  return new Int32Array([Number(excess), Number(bigPart)])
+}
+function bigIntToUint32Array(num: bigint): Uint32Array {
+  const maxInt = BigInt(Number.MAX_SAFE_INTEGER)
+  const excess = num % maxInt
+  const bigPart = num / maxInt
+  return new Uint32Array([Number(excess), Number(bigPart)])
+}
+
```

Comment:
Can you add short docstrings explaining how these work and why they're used? (Specifically, I'm curious why the arrays are constructed as `[least_significant, most_significant]` rather than vice-versa? Is this just how Arrow represents bignums internally?)

Would it be useful if they had their own unit tests, or are they already effectively tested by the ArrowVegaLiteChart tests? 

(Also teeny nit: there should be a space between the funcs.)